sys/include/sema.h

```diff
@@ -58,92 +78,76 @@ typedef struct {
  * @param[in] value Initial value for the semaphore.
  *
  * @return  0 on success.
- * @return  -EINVAL, if semaphore is invalid.
  */
 int sema_create(sema_t *sema, unsigned int value);
 
 /**
  * @brief   Destroys a semaphore.
  *
+ * @pre `(sema != NULL)`
+ *
  * @see <a href="http://pubs.opengroup.org/onlinepubs/9699919799/functions/sem_destroy.html">
  *          The Open Group Base Specifications Issue 7, sem_destroy()
  *      </a>
  *
+ * Destroying a semaphore upon which other threads are currently blocked
+ * will wake the other threads causing the @ref sema_wait (or
+ * @ref sema_wait_timed) to return error (-ECANCELLED).
+ *
  * @param[in] sema  The semaphore to destroy.
  *
  * @return  0 on success.
- * @return  -EINVAL, if semaphore is invalid.
  */
 int sema_destroy(sema_t *sema);
```

Comment:
I left it to keep the old return values meaningful. This implementation can't fail here so the return value makes no difference, but the same applies to the creation of the semaphore.

I'm happy to change it to void... I've no strong opinion about this one... your call.